substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -2160,11 +2160,15 @@ pub mod env {
 
 		let message: VersionedXcm<()> =
 			ctx.read_sandbox_memory_as_unbounded(memory, msg_ptr, msg_len)?;
-		let weight = <<E::T as Config>::Xcm as SendController<_>>::WeightInfo::send();
+		let weight = <<E::T as Config>::Xcm as SendController<_>>::WeightInfo::send_blob();
 		ctx.charge_gas(RuntimeCosts::CallRuntime(weight))?;
 		let origin = crate::RawOrigin::Signed(ctx.ext.address().clone()).into();
 
-		match <<E::T as Config>::Xcm>::send(origin, dest.into(), message.into()) {
+		match <<E::T as Config>::Xcm>::send_blob(
+			origin,
+			dest.into(),
+			message.encode().try_into().expect("TODO: What to do here?"),
```

Comment:
@pgherveou Any idea?

Reply:
Will look into it shortly feel free to leave a todo if that's unblock you

Reply:
We should adjust the interface here, we should read a BoundedVec<u8, MaxXcmEncodedSize> instead of VersionedXcm<()>

I can give it a shot and push a commit to your PR, early next week